### PR TITLE
Remove some extra static arrays used in Split()

### PR DESCRIPTION
**Pull request by iSazonov**

### CODE REVIEW

File: src/libraries/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs [hunk 1]
```diff
@@ -15,8 +15,6 @@ namespace Microsoft.CSharp
 {
     internal sealed partial class CSharpCodeGenerator : ICodeCompiler, ICodeGenerator
     {
-        private static readonly char[] s_periodArray = new char[] { '.' };
```
Comment (stephentoub) [inline src/libraries/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs:18]: Please revert all changes to files except the two System.Net ones. The other files are in projects that build for netstandard2.0 where the single char overload doesn't exist.

Comment (iSazonov) [reply src/libraries/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs]: Done.

File: src/libraries/System.Net.Mail/src/System/Net/Mime/MimeBasePart.cs [hunk 2]
```diff
@@ -43,7 +43,7 @@ internal static string EncodeHeaderValue(string value, Encoding? encoding, bool
         }
 
         private static readonly char[] s_headerValueSplitChars = new char[] { '\r', '\n', ' ' };
-        private static readonly char[] s_questionMarkSplitChars = new char[] { '?' };
+        private const char QuestionMarkSplitChar = '?';
```
Comment (stephentoub) [inline src/libraries/System.Net.Mail/src/System/Net/Mime/MimeBasePart.cs:46]:
```suggestion
```

File: src/libraries/System.Net.Mail/src/System/Net/Mime/MimeBasePart.cs [hunk 3]
```diff
@@ -65,7 +65,7 @@ internal static string DecodeHeaderValue(string? value)
                 //the third is the unicode encoding type, and the fourth is encoded message itself.  '?' is not valid inside of
                 //an encoded string other than as a separator for these five parts.
                 //If this check fails, the string is either not encoded or cannot be decoded by this method
-                string[] subStrings = foldedSubString.Split(s_questionMarkSplitChars);
+                string[] subStrings = foldedSubString.Split(QuestionMarkSplitChar);
```
Comment (stephentoub) [inline src/libraries/System.Net.Mail/src/System/Net/Mime/MimeBasePart.cs:68]:
```suggestion
                string[] subStrings = foldedSubString.Split('?');
```

File: src/libraries/System.Net.Mail/src/System/Net/Mime/MimeBasePart.cs (hunk 2 above)
Comment (iSazonov) [reply src/libraries/System.Net.Mail/src/System/Net/Mime/MimeBasePart.cs]: Done.

**Review by stephentoub — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**